packages/react-reconciler/src/ReactFiberNewContext.new.js

```diff
@@ -132,9 +137,17 @@ export function popProvider(
   const currentValue = valueCursor.current;
   pop(valueCursor, providerFiber);
   if (isPrimaryRenderer) {
-    context._currentValue = currentValue;
+    if (currentValue === REACT_SERVER_CONTEXT_DEFAULT_VALUE_NOT_LOADED) {
+      context._currentValue = ((context: any): ReactServerContext<any>)._defaultValue;
```

Comment:
This doesn't need refining neither. It could in theory be useful for the client context too in the future. So it's not wrong to apply it to both. Also helps us stay sound(er).